### ISSUE TITLE
error when run train_cnn.py

### ISSUE THREAD

**Issue by HebeLyu** [munhouiani/Deep-Packet#34]
when I tried to train the model on my own dataset, there was an error as follow:
 File "/home/inspur/lvzhuo/metrics learning/MultiModel_Plus/compare/deep-packet/train_cnn.py", line 25, in main
    train_application_classification_cnn_model(data_path, model_path,cls_num)
  File "/home/inspur/lvzhuo/metrics learning/MultiModel_Plus/compare/deep-packet/ml/utils.py", line 289, in train_application_classification_cnn_model
    train_cnn(
  File "/home/inspur/lvzhuo/metrics learning/MultiModel_Plus/compare/deep-packet/ml/utils.py", line 230, in train_cnn
    trainer.fit(model)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 603, in fit
    call._call_and_handle_interrupt(
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/call.py", line 38, in _call_and_handle_interrupt
    return trainer_fn(*args, **kwargs)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 645, in _fit_impl
    self._run(model, ckpt_path=self.ckpt_path)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1098, in _run
    results = self._run_stage()
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1177, in _run_stage
    self._run_train()
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1200, in _run_train
    self.fit_loop.run()
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/loops/loop.py", line 194, in run
    self.on_run_start(*args, **kwargs)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/loops/fit_loop.py", line 206, in on_run_start
    self.trainer.reset_train_dataloader(self.trainer.lightning_module)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1515, in reset_train_dataloader
    self.train_dataloader = self._data_connector._request_dataloader(RunningStage.TRAINING)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/data_connector.py", line 446, in _request_dataloader
    dataloader = source.dataloader()
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/data_connector.py", line 520, in dataloader
    return self.instance.trainer._call_lightning_module_hook(self.name, pl_module=self.instance)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/pytorch_lightning/trainer/trainer.py", line 1342, in _call_lightning_module_hook
    output = fn(*args, **kwargs)
  File "/home/inspur/lvzhuo/metrics learning/MultiModel_Plus/compare/deep-packet/ml/utils.py", line 111, in train_dataloader
    dataset_dict = datasets.load_dataset(self.hparams.data_path)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/load.py", line 1769, in load_dataset
    ds = builder_instance.as_dataset(split=split, ignore_verifications=ignore_verifications, in_memory=keep_in_memory)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/builder.py", line 1066, in as_dataset
    datasets = map_nested(
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 444, in map_nested
    mapped = [
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 445, in <listcomp>
    _single_map_nested((function, obj, types, None, True, None))
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/utils/py_utils.py", line 346, in _single_map_nested
    return function(data_struct)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/builder.py", line 1097, in _build_single_dataset
    ds = self._as_dataset(
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/builder.py", line 1168, in _as_dataset
    dataset_kwargs = ArrowReader(cache_dir, self.info).read(
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/arrow_reader.py", line 239, in read
    return self.read_files(files=files, original_instructions=instructions, in_memory=in_memory)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/arrow_reader.py", line 260, in read_files
    pa_table = self._read_files(files, in_memory=in_memory)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/arrow_reader.py", line 203, in _read_files
    pa_table = concat_tables(pa_tables) if len(pa_tables) != 1 else pa_tables[0]
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/table.py", line 1778, in concat_tables
    return ConcatenationTable.from_tables(tables, axis=axis)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/table.py", line 1484, in from_tables
    return cls.from_blocks(blocks)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/table.py", line 1427, in from_blocks
    table = cls._concat_blocks(blocks, axis=0)
  File "/home/inspur/anaconda3/envs/mulmodel/lib/python3.8/site-packages/datasets/table.py", line 1373, in _concat_blocks
    return pa.concat_tables(pa_tables, promote=True)
  File "pyarrow/table.pxi", line 5120, in pyarrow.lib.concat_tables
  File "pyarrow/error.pxi", line 144, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 100, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Unable to merge: Field feature has incompatible types: list<element: double> vs list<item: double>
  0%|          | 0/1 [00:01<?, ?it/s]

**Comment by munhouiani** [comment]
Do you have any raw data that is shareable? This is a `pyarrow`'s error. I'm not quite sure what the problem is.
